src/common/commonutils/FileUtils.c

```diff
@@ -120,25 +120,52 @@ bool SavePayloadToFile(const char* fileName, const char* payload, const int payl
 
 bool AppendPayloadToFile(const char* fileName, const char* payload, const int payloadSizeBytes, void* log)
 {
-    char* fileContents = NULL;
+    FILE* existingFile = NULL;
+    struct stat st;
     bool result = false;
+    char lastCharacter = EOF;
+
+    if(NULL == fileName || NULL == payload || 0 == payloadSizeBytes)
+    {
+        OsConfigLogError(log, "AppendPayloadToFile: invalid arguments");
+        return false;
+    }
 
-    // If the file exists and there is no EOL at the end of file, add one before the append
-    if ((NULL != payload) && (payloadSizeBytes > 0) && FileExists(fileName) && 
-        (NULL != (fileContents = LoadStringFromFile(fileName, false, log))) && 
-        (EOL != fileContents[strlen(fileContents) - 1]))
+    /* If the file exists and there is no EOL at the end of file, add one before the append. */

```

Comment:
>/* If the f [](http://example.com/codeflow?start=4&length=11)

Keep consistent with commenting style for this project. We use C++ style comments.

Reply:
That MI header is ingested from Windows as-is. This and the auto-generated NRP infra code use a different coding style from rest of our project.